### PR TITLE
Fix for issue

### PR DESCRIPTION
**Pull request by XxSNIPER2000xX**

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -186,9 +200,13 @@ static int parse_opt(int key, char *arg, struct argp_state *state) {
                 return 1;
             }
             break;
+        case 'p':
+            preload_entry();
+            break;
         case ARGP_KEY_ARG:
             if (state->arg_num == 0) {
                 // First argument is provided
+                // printf("File: %s\n", arg);
```
Comment (Zeyu-Li) [inline src/main.c]: can you remove this line

File: src/main.c [hunk 2]
```diff
@@ -211,9 +229,16 @@ int main(int argc, char *argv[]) {
     // no buffer when printing
     setbuf(stdout, NULL);
 
+    // if no flags
```
Comment (Zeyu-Li) [inline src/main.c]: can this part as well

File: src/preload.h [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+} Tape;
+
+void init_tape(Tape *t, size_t initSize) {
+    t->data = (uint8_t *) malloc(initSize * sizeof(uint8_t));
+    t->size = 0;
+}
+
+void __tape_resize__(Tape *t, size_t *len) { /* should be used by dev only */
+    uint8_t *resize = (uint8_t *) realloc(t->data, (size_t) 16 * sizeof(uint8_t));
+    if (resize == NULL) {
```
Comment (Zeyu-Li) [inline src/preload.h]: why is there logic in header file?

File: src/main.c [hunk 4]
```diff
@@ -172,6 +174,18 @@ void cin_entry() {
     }
 }
 
+void preload_entry() {
+    Tape t;
+    size_t initTapeSize = 8; // 8 bytes
+    init_tape(&t, initTapeSize);
+    fill_tape(&t, initTapeSize, "test2.txt");
```
Comment (Zeyu-Li) [inline src/main.c]: why is test2.txt hard coded here?

**Review by Zeyu-Li — CHANGES_REQUESTED** [review]

**Comment by Zeyu-Li** [comment]
Also add to makefile

File: src/main.c (hunk 4 above)
Comment (XxSNIPER2000xX) [reply src/main.c]: forgot to change that, test2.txt should be the argument for the file

**Pull request closed without merging**